mlflow/resource_model.go

```diff
@@ -12,16 +12,28 @@ type Tag struct {
 	Value string `json:"value"`
 }
 
+// ModelVersion defines a MLFlow model version as returned by the API
+type ModelVersion struct {
+	Name                 string `json:"name"`
+	Version              string `json:"version"`
+	CreationTimestamp    int64  `json:"creation_timestamp,omitempty"`
+	LastUpdatedTimestamp int64  `json:"last_updated_timestamp,omitempty"`
+	UserID               string `json:"user_id,omitempty"`
+	CurrentStage         string `json:"current_stage,omitempty"`
+	Source               string `json:"source,omitempty"`
+	Status               string `json:"status,omitempty"`
+}
+
 // Model defines the response object from the API
 type Model struct {
-	Name                 string   `json:"name" tf:"force_new"`
-	CreationTimestamp    int64    `json:"creation_timestamp,omitempty" tf:"computed"`
-	LastUpdatedTimestamp int64    `json:"last_updated_timestamp,omitempty" tf:"computed"`
-	UserID               string   `json:"user_id,omitempty" tf:"computed"`
-	LatestVersions       []string `json:"latest_versions,omitempty" tf:"computed"`
-	Description          string   `json:"description,omitempty"`
-	Tags                 []Tag    `json:"tags,omitempty"`
-	RegisteredModelID    string   `json:"id,omitempty" tf:"computed,alias:registered_model_id"`
+	Name                 string         `json:"name" tf:"force_new"`
+	CreationTimestamp    int64          `json:"creation_timestamp,omitempty" tf:"computed"`
+	LastUpdatedTimestamp int64          `json:"last_updated_timestamp,omitempty" tf:"computed"`
+	UserID               string         `json:"user_id,omitempty" tf:"computed"`
+	LatestVersions       []ModelVersion `json:"latest_versions,omitempty" tf:"computed"`
```

Comment:
let me check if this won't proliferate into tfstate, otherwise i'll add ignore for `latest_versions` in schema customization function.